lib/mongodb/mongodb.nit

```diff
@@ -0,0 +1,650 @@
+	# var client = new MongoClient("mongodb://localhost:27017/")
+	# var col = client.database("test").collection("test")
+	# var sel = new JsonObject
+	# sel["foo"] = 10
+	# var upd = new JsonObject
+	# upd["bar"] = "BAR"
+	# assert col.update(sel, upd)
+	# ~~~
+	fun update(selector: JsonObject, update: JsonObject): Bool do
+		assert is_alive
+		return native_instance.update(
+			selector.to_bson.native_instance,
+			update.to_bson.native_instance)
+	end
+
+	# Updates all documents matching the `selector`.
+	#
+	# See `update`.
+	fun update_all(selector: JsonObject, update: JsonObject): Bool do
+		return native_instance.update_all(
+			selector.to_bson.native_instance,
+			update.to_bson.native_instance)
+	end
+
+	# Counts the document matching `query`.
+	#
+	# Returns `-1` if an error occured. See `Sys::last_mongoc_error`.
+	#
+	# ~~~
+	# var client = new MongoClient("mongodb://localhost:27017/")
+	# var col = client.database("test").collection("test")
+	# var query = new JsonObject
+	# query["foo"] = 10
+	# assert col.count(query) > 0
+	# ~~~
+	fun count(query: JsonObject): Int do
+		assert is_alive
+		return native_instance.count(query.to_bson.native_instance)
+	end
+
+	# Finds the first document that matches `query`.
+	#
+	# Returns `null` if an error occured. See `Sys::last_mongoc_error`.
+	#
+	# ~~~
+	# var client = new MongoClient("mongodb://localhost:27017/")
+	# var col = client.database("test").collection("test")
+	# var query = new JsonObject
+	# query["foo"] = 10
+	# var doc = col.find(query)
+	# print doc or else "nullllllllllllllllll"
+	# print doc.to_json
+	# assert doc["foo"] == 10
+	# ~~~
+	fun find(query: JsonObject): nullable JsonObject do
+		assert is_alive
+		var c = native_instance.find(query.to_bson.native_instance)
+		if c == null then return null
+		var cursor = new MongoCursor(c)
+		if cursor.is_ok then
+			cursor.next
+			return cursor.item
+		end
+		return null
+	end
+
+	# Finds all the documents matching the `query`.
+	#
+	# ~~~
+	# var client = new MongoClient("mongodb://localhost:27017/")
+	# var col = client.database("test").collection("test")
+	# var query = new JsonObject
+	# query["foo"] = 10
+	# assert col.find_all(query).length > 0
+	# ~~~
+	fun find_all(query: JsonObject): Array[JsonObject] do
+		assert is_alive
+		var res = new Array[JsonObject]
+		var c = native_instance.find(query.to_bson.native_instance)
+		if c == null then return res
+		var cursor = new MongoCursor(c)
+		for item in cursor do res.add item
+		return res
+	end
+
+	# Retrieves statistics about the collection.
+	#
+	# Returns `null` if an error occured. See `Sys::last_mongoc_error`.
+	#
+	# ~~~
+	# var client = new MongoClient("mongodb://localhost:27017/")
+	# var col = client.database("test").collection("test")
+	# assert col.stats["ns"] == "test.test"
+	# ~~~
+	fun stats: nullable JsonObject do
+		assert is_alive
+		var bson = native_instance.stats
+		if bson == null then return null
+		return new JsonObject.from_bson(new BSON(bson))
```

Comment:
`bson` could be freed here. If this is the only user of `Native...stats` then it would be (almost) enough...